optimum/onnxruntime/quantization.py

```diff
@@ -0,0 +1,262 @@
+#  Copyright 2021 The HuggingFace Team. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+from enum import Enum
+from pathlib import Path
+from typing import Callable, Optional, Union
+
+import numpy
+import torch
+from datasets import Dataset, load_dataset
+from torch.utils.data import DataLoader, RandomSampler
+from transformers import AutoTokenizer, default_data_collator
+from transformers.onnx import export
+from transformers.onnx.features import FeaturesManager
+
+import onnx
+from onnxruntime.quantization import (
+    CalibrationDataReader,
+    CalibrationMethod,
+    QuantFormat,
+    QuantType,
+    quantize_dynamic,
+    quantize_static,
+)
+from optimum.onnxruntime.configuration import ORTConfig
+from optimum.onnxruntime.utils import generate_identified_filename
+
+
+logger = logging.getLogger(__name__)
+
+
+class ORTQuantizationMode(Enum):
+
+    DYNAMIC = "dynamic"
+    STATIC = "static"
+
+
+SUPPORTED_QUANT_MODE = set([approach.value for approach in ORTQuantizationMode])
+
+CALIB_METHOD = {"minmax": CalibrationMethod.MinMax, "entropy": CalibrationMethod.Entropy}
+
+Q_FORMAT = {"operator": QuantFormat.QOperator, "qdq": QuantFormat.QDQ}
+
+Q_TYPE = {"int8": QuantType.QInt8, "uint8": QuantType.QUInt8}
+
+
+class ORTCalibrationDataReader(CalibrationDataReader):
+    def __init__(self, calib_dataloader: DataLoader):
+        self._iter = iter([{key: data[key].numpy() for key in data} for data in calib_dataloader])
+
+    def get_next(self):
+        return next(self._iter, None)
+
+
+class ORTQuantizer:
+    def __init__(
+        self,
+        model_name_or_path: str,
```

Comment:
What about `Union[str, os.PathLike]` here? people will get the sense this two paths are distincts